ob1k-cache/src/main/java/com/outbrain/ob1k/cache/memcache/MemcacheClient.java

```diff
@@ -94,6 +94,11 @@ private static int expirationInSpyUnits(final long exp, final TimeUnit unit) {
   }
 
   @Override
+  public ComposableFuture<Boolean> putIfAbsentAsync(final K key, final V value) {
+    throw new UnsupportedOperationException("Not supported for spy");
```

Comment:
are you sure spy doesn't support for add operation? better to implement this also if we can :)